sql-common/my_time.c

```diff
@@ -336,12 +339,25 @@ str_to_datetime(const char *str, uint length, MYSQL_TIME *l_time,
     pos++;
     skip_digits(&pos, end); // ignore the return value
   }
+  if (pos < end && *pos == '-' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu-HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
+  }
+  else if (pos < end && *pos == '+' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuu+HH:MM is supported too */
+  {
+    pos+=6; // Skip 6 places as defined in RFC3339 reserved for local offset
+  }
+  else if (pos < end && *pos == 'Z' && digits >= 12) /* YYYYYMMDDHHMMSShhmmss.uuuuuuZ is supported too */
+  {
+    pos++;
+  }
+
 
   if (pos == end)
   {
     /*
       Found date in internal format
-      (only numbers like [YY]YYMMDD[T][hhmmss[.uuuuuu]])
+      (only numbers like [YY]YYMMDD[T][hhmmss[.uuuuuu]][[Z][[-][+]hh:mm]])
```

Comment:
Is extending this a little to the ISO8601 formats (excluding week and ordinal formats) of the following possible?
<code>
[YY]YYMMDD[T|t][hhmmss[.uuuuuu]][Z|z]|[-|+]hh[[:]mm]]
</code>
So the ``:`` and the ``mm`` in the time are optional.

Believe me I know how fiddly date parsing is (https://github.com/fail2ban/fail2ban/commits/0.10/fail2ban/server/datedetector.py?author=grooverdan)